changelog/0032-upgrade-elasticsearch.yml

```diff
@@ -0,0 +1,76 @@
+---
+title: ES upgrade from 1.7.6 to 2.4.6
+key: upgrade-elasticsearch
+date: 2020-02-05
+optional_per_env: yes
+update_steps: |
+
+  1. Stop the site
+  ```bash
+  commcare-cloud <env> downtime start
+  ```
+    
+  2. Create a PR by adding below parameters in commcare-cloud/environments/<env>/public.yml and merge
+  { 
+  elasticsearch_version: 2.4.6
+  elasticsearch_download_sha256: 5f7e4bb792917bb7ffc2a5f612dfec87416d54563f795d6a70637befef4cfc6f.
+  ELASTICSEARCH_MAJOR_VERSION: 2
+  }
+    
+  3. Update the local settings
+  ```bash
+  commcare-cloud <env> update-config
+  ```  
+  4. Login to any ES server and disable the cluster routing
+  ```bash
+  curl -XPUT Elasticsearchserver-IP:9200/_cluster/settings -d '{\"persistent\" : {\"cluster.routing.allocation.enable\" : \"none\" }}'
+  ```
```

Comment:
I see now one of the commands below uses `<ES-IP>`, which looks good.  This should be consistent throughout the instructions.

Reply:
Throughout instead of `<ES-IP>`, you can use `$(cchq ${ENV} lookup elasticsearch:0)` which will actually calculate that. Since there are so many commands to run, I might also tell them to set their ENV environment variable, and then use `${ENV}` throughout these commands

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/2.4/restart-upgrade.html#_step_1_disable_shard_allocation_2 also suggests performing a `POST /_flush/synced` before shutting down.